components/brave_shields/browser/brave_shields_util.cc

```diff
@@ -443,10 +443,26 @@ void SetCookieControlType(HostContentSettingsMap* map,
       case ControlType::BLOCK:
         map->SetDefaultContentSetting(ContentSettingsType::COOKIES,
                                       CONTENT_SETTING_BLOCK);
+        // Toggle the state off->on to refresh the upstream Third-party cookies
+        // Settings page correctly. This is a temporary measure until we figure
+        // out a better UI for Cookies Settings page.
+        profile_state->SetInteger(
```

Comment:
it's not clear to me why this is necessary. If it's already on block 3p then it will continue to block 3p in addition to blocking 1p, right? Why does it need to toggle off first?

Reply:
switch between `ControlType::BLOCK` and `ControlType::BLOCK_THIRD_PARTY` won't touch `prefs::kCookieControlsMode` which lead to the radio group stay disabled when it should've become enabled.